src/meanshift.jl

```diff
@@ -0,0 +1,98 @@
+using Images, NearestNeighbors, Distances, StaticArrays, Clustering
+
+function meanshift(img::Array{Float64, 2}, spatial_radius::Float64, range_radius::Float64, min_density::Int)
```

Comment:
It wouldn't be difficult to extend the algorithm to N-dimensional images. With the current performance, I didn't think it would be useful to support N-dimensional images.